src/corehost/cli/fxr/sdk_resolver.cpp

```diff
@@ -5,183 +5,240 @@
-                // If a global cli version is specified:
-                //   pick the greatest version that differs only in the 'minor-patch'
-                //   and is semantically greater than or equal to the global cli version specified.
-                (ver.get_major() == specified.get_major() && ver.get_minor() == specified.get_minor() &&
-                (ver.get_patch() / 100) == (specified.get_patch() / 100) && ver >= specified))
-            {
-                max_ver = std::max(ver, max_ver);
-            }
-        }
+    if (name == _X("patch"))
+    {
+        return sdk_roll_forward_policy::patch;
     }
 
-    if (!max_ver.is_empty())
+    if (name == _X("feature"))
     {
-        pal::string_t max_ver_str = max_ver.as_str();
-        append_path(&sdk_path, max_ver_str.c_str());
+        return sdk_roll_forward_policy::feature;
+    }
 
-        trace::verbose(_X("Checking if resolved SDK dir [%s] exists"), sdk_path.c_str());
-        if (pal::directory_exists(sdk_path))
-        {
-            trace::verbose(_X("Resolved SDK dir is [%s]"), sdk_path.c_str());
-            retval = max_ver_str;
-        }
+    if (name == _X("minor"))
+    {
+        return sdk_roll_forward_policy::minor;
     }
 
-    return retval;
-}
+    if (name == _X("major"))
+    {
+        return sdk_roll_forward_policy::major;
+    }
 
-bool sdk_resolver_t::resolve_sdk_dotnet_path(const pal::string_t& dotnet_root, pal::string_t* cli_sdk)
-{
-    trace::verbose(_X("--- Resolving dotnet from working dir"));
-    pal::string_t cwd;
-    if (!pal::getcwd(&cwd))
+    if (name == _X("latestPatch"))
     {
-        trace::verbose(_X("Failed to obtain current working dir"));
-        assert(cwd.empty());
+        return sdk_roll_forward_policy::latest_patch;
     }
 
-    return resolve_sdk_dotnet_path(dotnet_root, cwd, cli_sdk);
-}
+    if (name == _X("latestFeature"))
+    {
+        return sdk_roll_forward_policy::latest_feature;
+    }
 
-bool higher_sdk_version(const pal::string_t& new_version, pal::string_t* version)
-{
-    bool disallow_prerelease = false;
-    bool retval = false;
-    fx_ver_t ver;
-    fx_ver_t new_ver;
+    if (name == _X("latestMinor"))
+    {
+        return sdk_roll_forward_policy::latest_minor;
+    }
 
-    if (fx_ver_t::parse(new_version, &new_ver, disallow_prerelease))
+    if (name == _X("latestMajor"))
     {
-        if (!fx_ver_t::parse(*version, &ver, disallow_prerelease) || (new_ver > ver))
-        {
-            version->assign(new_version);
-            retval = true;
-        }
+        return sdk_roll_forward_policy::latest_major;
     }
 
-    return retval;
+    return sdk_roll_forward_policy::unsupported;
 }
 
-bool sdk_resolver_t::resolve_sdk_dotnet_path(
-    const pal::string_t& dotnet_root, 
-    const pal::string_t& cwd, 
-    pal::string_t* cli_sdk,
-    bool disallow_prerelease,
-    pal::string_t* global_json_path)
+const char* sdk_resolver::to_policy_name(sdk_roll_forward_policy policy)
```

Comment:
I wasn't terribly worried about the optimization (might even get optimized out anyway) rather than making the logic pretty clear.  However, I agree that the redundant logic can be removed.  I'll fix that.